### DISCUSSION TITLE
Pixel perfect interaction, take into account the alpha

### DISCUSSION THREAD

**Discussion by Bouh** [pixijs/pixijs#7591] [Ideas]
# Current state

Currently, the interactions can be done on the whole Sprite or with a mask of polygons to define manually.

# Suggestion

Add support for the pixel-perfect interaction [like Phaser ](https://phaser.io/examples/v2/input/pixel-perfect-click-detection)can do.

# Question

There is a reason for that feature is missing from the beginning of Pixi?

I agree this is costly to check the pixels datas, but it a concient choose make by the user when we want use it.

**Comment by ivanpopelyshev** [comment]
Try make a PR, get rejected, make better, get rejected, and in 5 or 10 iterations we'll get somethin good enough to be merged :) That's the process for performance-heavy features. I went through that, its not pretty.

https://github.com/pixijs/pixijs/wiki/v5-Hacks#pixel-perfect-interaction

Or you can get lucky and make something good at first try.
